src/processors/processUpdateRequest.ts

```diff
@@ -0,0 +1,81 @@
+
+  const updatedRequest = processVehicleIdentifiers(recordFromDB, requestBody);
+
+  const newRecord = { ...formattedRecordFromDB, ...updatedRequest } as TechrecordGet;
+
+  newRecord.techRecord_recordCompleteness = validateAndComputeRecordCompleteness(newRecord as TechrecordPut);
+
+  const flattenedNewRecord = await flattenArrays(newRecord) as TechrecordGet;
+
+  const updateType = getUpdateType(flattenedNewRecord, recordFromDB);
+  recordFromDB.techRecord_updateType = updateType;
+
+  const date = new Date().toISOString();
+  const updatedRecordFromDB = setLastUpdatedAuditDetails(recordFromDB, userDetails.username, userDetails.msOid, date);
+  const updatedNewRecord = setCreatedAuditDetails(flattenedNewRecord, userDetails.username, userDetails.msOid, date);
+
+  return [updatedRecordFromDB, updatedNewRecord];
+};
+
+export const setLastUpdatedAuditDetails = (techRecord: TechrecordGet, createdByName: string, createdById: string, date: string) => {
+  techRecord.techRecord_lastUpdatedAt = date;
+  techRecord.techRecord_lastUpdatedByName = createdByName;
+  techRecord.techRecord_lastUpdatedById = createdById;
+  techRecord.techRecord_statusCode = STATUS.ARCHIVED;
+  return techRecord;
+};
+
+export const setCreatedAuditDetails = (techRecord: TechrecordGet, createdByName: string, createdById: string, date: string) => {
+  techRecord.techRecord_createdAt = date;
+  techRecord.techRecord_createdByName = createdByName;
+  techRecord.techRecord_createdById = createdById;
+  techRecord.createdTimestamp = date;
+  delete techRecord.techRecord_lastUpdatedAt;
+  delete techRecord.techRecord_lastUpdatedById;
+  delete techRecord.techRecord_lastUpdatedByName;
+  return techRecord as TechrecordPut;
+};
+
+export const getUpdateType = (oldRecord: TechrecordGet, newRecord: TechrecordGet): UpdateType => {
+  let updateType = UpdateType.TECH_RECORD_UPDATE;
+  Object.entries(newRecord).forEach(([key, value]) => {
+    if (/techRecord_adrDetails_[a-zA-Z]+/.test(key)) {
+      if (value !== oldRecord[key as keyof TechrecordGet]) {
+        updateType = UpdateType.ADR;
+        return updateType;
+      }
+    }
+    return updateType;
+  });
+  return updateType;
+};
+
+export const processVehicleIdentifiers = (recordFromDB: TechrecordGet, requestBody: TechrecordPut) => {
+  const techRecord = { ...requestBody } as TechrecordGet;
+  if ((techRecord as TechrecordHgv | TechrecordMotorcycle | TechrecordCar | TechrecordPsv).primaryVrm !== (recordFromDB as TechrecordHgv | TechrecordMotorcycle | TechrecordCar | TechrecordPsv).primaryVrm) {
+    (techRecord as TechrecordHgv | TechrecordMotorcycle | TechrecordCar | TechrecordPsv).primaryVrm = (recordFromDB as TechrecordHgv | TechrecordMotorcycle | TechrecordCar | TechrecordPsv).primaryVrm;
+  }
+  if ((techRecord as TechrecordTrl).trailerId !== (recordFromDB as TechrecordTrl).trailerId) {
+    (techRecord as TechrecordTrl).trailerId = (recordFromDB as TechrecordTrl).trailerId;
+  }
```

Comment:
As discussed, can we not just delete these vehicle Identifiers from the incoming request? For instance in the where we validate the request in the handler?

Reply:
With delete I am getting typescript error (The operand of a 'delete' operator must be optional.) because trailerId is not an optional field in trailer schemas.